avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroGenerator.java

```diff
@@ -382,6 +390,17 @@ public final void writeStartObject() throws IOException {
     }
 
     @Override
+    public void writeStartObject(Object forValue) throws IOException {
```

Comment:
I tried to maintain the old codepath as well, so if you call `writeStartObject()` instead of `writeStartObject(Object)` on a non-union, it shouldn't care.